program/src/processor.rs

```diff
@@ -1227,6 +1238,10 @@ fn process_update_governance(
     state.voting_period_seconds = voting_period_seconds;
     state.stake_per_proposal = stake_per_proposal;
 
+    if cooldown_seconds > 0 && (Clock::get()?.unix_timestamp as u64) < state.cooldown_expires {
+        state.cooldown_expires = Clock::get()?.unix_timestamp as u64 + cooldown_seconds;
+    }
+
```

Comment:
I don't think its necessary to reset the cooldown on governance update. As governance updates should only be possible by governance itself (which is disabled until cooldown expires).